xonsh/commands_cache.py

```diff
@@ -197,7 +197,8 @@ def _update_cmds_cache(
 
         for cmd, path in self._get_all_cmds(paths).items():
             key = cmd.upper() if ON_WINDOWS else cmd
-            allcmds[key] = (path, aliases.get(key, None))
+            if key not in allcmds:
+                allcmds[key] = (path, aliases.get(key, None))
```

Comment:
this means any override will not work. If Python is overridded by virtualenv, it will not have the updated path.